examples/bosd-payload-with-rust-bitcoin.rs

```diff
@@ -0,0 +1,108 @@
+//! Implements an example PSBT workflow.
+//!
+//! The workflow we simulate is that of a setup using a BOSD payload to create a PSBT with a P2WPKH output and a OP_return output.
+//!
+//! ## How to check:
+//!
+//! 1. Start Bitcoin Core in Regtest mode, for example:
+//!
+//!    `bitcoind -regtest -server -daemon -fallbackfee=0.0002 -rpcuser=admin -rpcpassword=pass -rpcallowip=127.0.0.1/0 -rpcbind=127.0.0.1 -blockfilterindex=1 -peerblockfilters=1`
+//!
+//! 2. Define a shell alias to `bitcoin-cli`, for example:
+//!
+//!    `alias bt="bitcoin-cli -rpcuser=admin -rpcpassword=pass -rpcport=18443"`
+//!
+//! 3. Run this example to get the Base64 encoded PSBT with:
+//!
+//!    `cargo run --features examples --example payload_with_rust-bitcoin`
+//!
+//! 4. Now that you have with the Base64 enconded PSBT:
+//!
+//!    `bt analyzepsbt cHNidP8BAEACAAAAAAIAAAAAAAAAABYAFJf1yVh7sF47WTYWjYsqsoGUbuOsAAAAAAAAAAAOagxCaXRjb2luIEJPU0QAAAAAAAAA`
+//!
+
+#[cfg(feature = "examples-dep")]
+use base64::{engine::general_purpose::STANDARD, Engine};
+
+use bitcoin::{
+    absolute::LockTime,
+    bip32::{IntoDerivationPath, Xpriv},
+    key::Secp256k1,
+    secp256k1::All,
+    transaction::Version,
+    Amount, Network, Psbt, Transaction, TxOut,
+};
+
+fn main() {
+    // This example shows the basic usage of the [`bitcoin_bosd::Descriptor`] struct
+    // which is responsible for parsing and transmuting BOSD payloads into Bitcoin [`ScriptBuf`] and [`Address`].
+
+    // Example of a OP_RETURN payload.
+    let message = String::from("Bitcoin BOSD");
+    let payload = [&[0u8; 1], message.as_bytes()].concat(); // The BOSD payload is simply the type tag followed by the cryptographic payload.
+
+    let opreturn_desc = bitcoin_bosd::Descriptor::from_bytes(payload.as_slice()).unwrap();
+
+    let op_return_payload_script = opreturn_desc.to_script();
+
+    // Here we use the is_op_return() method from rust-bitcoin to check if the script is indeed an op_return.
+    assert!(op_return_payload_script.is_op_return());
+    // [`Descriptor`]s can be derived to a Bitcoin [`Address`].
+
+    // This will result in a error since this specific one is an OP_RETURN.
+    let op_return_address_error = opreturn_desc.to_address(Network::Bitcoin);
+    assert_eq!(
+        op_return_address_error.err().unwrap(),
+        bitcoin_bosd::DescriptorError::InvalidAddressConversion(
+            bitcoin_bosd::DescriptorType::OpReturn
+        )
+    );
+
+    //  Example of a P2WPKH payload.
+    let payload: [u8; 20] = [
+        // random hash160
+        0x97, 0xf5, 0xc9, 0x58, 0x7b, 0xb0, 0x5e, 0x3b, 0x59, 0x36, 0x16, 0x8d, 0x8b, 0x2a, 0xb2,
+        0x81, 0x94, 0x6e, 0xe3, 0xac,
+    ];
+    //  First byte is 3 to indicate a P2WPKH hash.
```

Comment:
```suggestion
    //  First byte is 3 to indicate a SegWit V0 type tag.
```
Check the spec in [SPECIFICATION.md]